cogito/put.go

```diff
@@ -277,14 +278,32 @@ type gitURL struct {
 	Repo  string
 }
 
+// safeUrlParse wraps [url.Parse] and returns only the error and not the URL to avoid leaking
+// passwords of the form http://user:password@example.com
+//
+// From https://github.com/golang/go/issues/53993
+func safeUrlParse(rawURL string) (*url.URL, error) {
+	parsedUrl, err := url.Parse(rawURL)
+	if err != nil {
+		if uerr := new(url.Error); errors.As(err, &uerr) {
```

Comment:
I realized that this code, although correct, can be confusing.
The idiomatic way to use errors.As is
```
		var uerr *url.Error
		if errors.As(err, &uerr) {
```
@iAmoric  Could you please update?

See https://go.dev/blog/go1.13-errors

Reply:
Fixed in [c544b0c](https://github.com/Pix4D/cogito/pull/94/commits/c544b0c86c4f8398ac6ce008fdacfa21a0ef9f0f)